toast-tk-runtime/src/main/java/io/toast/tk/runtime/mail/SmtpConfigProvider.java

```diff
@@ -0,0 +1,114 @@
+import java.io.Reader;
+import java.security.Security;
+import java.util.Properties;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.toast.tk.runtime.constant.Property;
+import io.toast.tk.runtime.utils.EncryptHelper;
+
+public class SmtpConfigProvider {
+
+	private static final Logger LOG = LogManager.getLogger(SmtpConfigProvider.class);
+
+	private String proxyURL;
+	private String proxyPort;
+	
+	private Properties properties;
+	
+	public static final String SMTP_PROPERTIES_FILE_NAME = "smtp.properties";
+	public static final String SMTP_PROPERTIES_FILE_STREAM_PATH = "/" + SMTP_PROPERTIES_FILE_NAME;
+	public static final String SMTP_PROPERTIES_FILE_PATH = Property.TOAST_HOME_DIR + SMTP_PROPERTIES_FILE_NAME;
+	
+	public static final String SMTP_PROPERTIES_HOST = "mail.smtp.host";
+
+	public static final String SMTP_PROPERTIES_PORT = "mail.smtp.port";
+
+	public static final String SMTP_PROPERTIES_TTLS_ENABLE = "mail.smtp.starttls.enable";
+
+	public static final String SMTP_PROPERTIES_AUTH = "mail.smtp.auth";
+
+	public static final String SMTP_PROPERTIES_USER = "mail.smtp.user";
+
+	public static final String SMTP_PROPERTIES_PASSWORD = "mail.smtp.password";
+	
+	public SmtpConfigProvider() {
+		properties = new Properties();
+		this.init();
+	}
+
+	public SmtpConfigProvider(String host, String port, String user, String pswd) {
+		properties = new Properties();
+		this.init(host, port, user, pswd);
+	}
+	
+	public void setProxy(String proxyURL, String proxyPort) {
+		this.proxyURL = proxyURL;
+		this.proxyPort = proxyPort;
+	}
+	
+	public void init() {
+		InputStream resourceAsStream = this.getClass().getResourceAsStream(SMTP_PROPERTIES_FILE_STREAM_PATH);
+
+		if(resourceAsStream == null) {
+			try {
+				properties.load(new FileReader(SMTP_PROPERTIES_FILE_PATH));
+			}
+			catch(IOException e) {
+				LOG.error(e.getMessage(), e);
+			}
+		} else {
+			try (final Reader resourceFileReader = new InputStreamReader(resourceAsStream)) {
+				properties.load(resourceFileReader);
+			} catch (final IOException e) {
+				LOG.error("Could not load smtp.properties file", e);
+			}
+		}
+
+		properties.put(SMTP_PROPERTIES_PASSWORD, EncryptHelper.decrypt((String) properties.get(SMTP_PROPERTIES_PASSWORD)));
+
+		setProperty(properties);
+	}
+	
+	public void init(String host, String port, String user, String pswd) {
+		properties.setProperty(SMTP_PROPERTIES_HOST, host);
+		properties.setProperty(SMTP_PROPERTIES_PORT, port);
+		if(user != null || "".equals(user)) {
+			properties.setProperty(SMTP_PROPERTIES_AUTH, "true");
+			properties.setProperty(SMTP_PROPERTIES_USER, user);
+			properties.setProperty(SMTP_PROPERTIES_PASSWORD, pswd);
+		}
+
+		setProperty(properties);
+	}
+	
+	public final Properties get() {
+		return properties;
+	}
+	
+	@SuppressWarnings("restriction")
+	private void setProperty(Properties properties) {
+
+        Security.addProvider(new com.sun.net.ssl.internal.ssl.Provider());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use classes from the Java API instead of Sun classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1191)